File: Phanfare.ExternalAPI/PublicProfile.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml;

namespace Phanfare.ExternalAPI
{
	public class PublicProfile
	{
		public long UserID { get; set; }
		public string FirstName { get; set; }
		public string LastName { get; set; }
		public string Relation { get; set; } // self, friend, family
		public bool IsPremium { get; set; }
		public string ImageURL { get; set; }

		public static PublicProfile FromXML( XmlElement el )
		{
			PublicProfile profile = new PublicProfile();
			profile.UserID = el.GetAttributeInt64( "uid" );
			profile.FirstName = el.GetAttribute( "first_name" );
			profile.LastName = el.GetAttribute( "last_name" );
			profile.Relation = el.GetAttribute( "relation" );
			profile.IsPremium = el.GetAttributeBoolean( "premium" );
			profile.ImageURL = el.GetAttribute( "image_url" );
			return profile;
		}
	}
}

File: trunk/PhanfareJavaAPI/src/com/phanfare/api/PublicProfile.java
package com.phanfare.api;

import org.json.me.JSONException;
import org.json.me.JSONObject;

public class PublicProfile {
	public final static int RELATION_OTHER = 0;
	public final static int RELATION_SELF = 1;
	public final static int RELATION_FRIEND = 2;
	public final static int RELATION_FAMILY = 3;

	public long userId;
	public String firstName;
	public String lastName;
	public int relation;
	public boolean isPremium;
	public String imageUrl;

	public static PublicProfile fromJson(JSONObject object) {
		PublicProfile profile = new PublicProfile();
		try {
			profile.userId = object.getLong("uid");
			profile.firstName = object.getString("first_name");
			profile.lastName = object.getString("last_name");
			String relation = object.getString("relation").toLowerCase();
			if (relation.equals("self") == true)
				profile.relation = PublicProfile.RELATION_SELF;
			else if (relation.equals("friend") == true)
				profile.relation = PublicProfile.RELATION_FRIEND;
			else if (relation.equals("family") == true)
				profile.relation = PublicProfile.RELATION_FAMILY;
			else
				profile.relation = PublicProfile.RELATION_OTHER;
			profile.isPremium = object.getBoolean("premium");
			profile.imageUrl = object.getString("image_url");
		} catch (JSONException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
			return null;
		}
		return profile;
	}
}
